packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -71,12 +106,266 @@ export class IamPolicyChecksWebview extends VueWebview {
+                                data.findings.forEach((finding: AccessAnalyzer.ValidatePolicyFinding) => {
+                                    const message = `${finding.findingType}: ${finding.issueCode} - ${finding.findingDetails} Learn more: ${finding.learnMoreLink}`
+                                    if (finding.findingType === 'ERROR') {
+                                        diagnostics.push(
+                                            new vscode.Diagnostic(
+                                                new vscode.Range(
+                                                    finding.locations[0].span.start.line,
+                                                    finding.locations[0].span.start.offset,
+                                                    finding.locations[0].span.end.line,
+                                                    finding.locations[0].span.end.offset
+                                                ),
+                                                message,
+                                                vscode.DiagnosticSeverity.Error
+                                            )
+                                        )
+                                        validatePolicyDiagnosticCollection.set(
+                                            IamPolicyChecksWebview.editedDocumentUri,
+                                            diagnostics
+                                        )
+                                    } else {
+                                        diagnostics.push(
+                                            new vscode.Diagnostic(
+                                                new vscode.Range(
+                                                    finding.locations[0].span.start.line,
+                                                    finding.locations[0].span.start.offset,
+                                                    finding.locations[0].span.end.line,
+                                                    finding.locations[0].span.end.offset
+                                                ),
+                                                message,
+                                                vscode.DiagnosticSeverity.Warning
+                                            )
+                                        )
+                                        validatePolicyDiagnosticCollection.set(
+                                            IamPolicyChecksWebview.editedDocumentUri,
+                                            diagnostics
+                                        )
+                                    }
+                                })
+                                this.onValidatePolicyResponse.fire([
+                                    'Please view the problems panel to review the issues with your policy document. Policy checks should be run until issues are no longer found in your policy document.',
```

Comment:
nit: recommend moving the static texts to a constants file. 

Also, we should review with the doc writer for the final custom text messages for errors

Reply:
(we can finalize the error strings later, please track it in your ticket)